src/main/java/ay2122s1_cs2103t_w16_2/btbb/storage/JsonAdaptedIngredient.java

```diff
@@ -58,12 +58,12 @@ public Ingredient toModelType() throws IllegalValueException {
         // Quantity:
         if (quantity == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
-                    Quantity.class.getSimpleName()));
+                    GenericQuantity.class.getSimpleName()));
         }
-        if (!Quantity.isValidQuantity(quantity)) {
-            throw new IllegalValueException(Quantity.MESSAGE_CONSTRAINTS);
+        if (!GenericQuantity.isValidInternalQuantity(quantity)) {
+            throw new IllegalValueException(GenericQuantity.getMessageConstraints("Ingredient"));
```

Comment:
I think this should be quantity
```suggestion
            throw new IllegalValueException(GenericQuantity.getMessageConstraints("Quantity"));
```